src/components/common/HeaderNavigation.tsx

```diff
@@ -0,0 +1,166 @@
+  () => import('@graasp/ui').then((mod) => mod.PlatformSwitch),
+  {
+    ssr: false,
+  },
+);
+
+// TODO: use graasp-ui when window conflict is resolved
+// meanwhile we copy the definitions here
+export enum Platform {
+  Builder = 'Builder',
+  Player = 'Player',
+  Library = 'Library',
+  Analytics = 'Analytics',
+}
+
+// TODO: use graasp-ui when window conflict is resolved
+// meanwhile we copy the definitions here
+export type HostsMapper = Partial<
+  Record<Platform, (itemId?: string) => string | undefined>
+>;
+
+// TODO: use graasp-ui when window conflict is resolved
+// meanwhile we copy the definitions here
+export function defaultHostsMapper(
+  hostsUrls: Partial<Record<Platform, string>>,
+): HostsMapper {
+  const urlBuilders = {
+    [Platform.Builder]: (origin: string, itemId: string) =>
+      `${origin}/items/${itemId}`,
+    [Platform.Player]: (origin: string, itemId: string) =>
+      `${origin}/${itemId}`,
+    [Platform.Library]: (origin: string, itemId: string) =>
+      `${origin}/collections/${itemId}`,
+    [Platform.Analytics]: (origin: string, itemId: string) =>
+      `${origin}/${itemId}`,
+  };
+
+  return Object.fromEntries(
+    Object.entries(hostsUrls).map(([platform, url]) => {
+      const { origin } = new URL(url);
+      return [
+        platform,
+        // if passed itemId is undefined, redirect to home page of platform
+        (itemId: string) =>
+          itemId ? urlBuilders[platform as Platform](origin, itemId) : origin,
+      ];
+    }),
+  ) as HostsMapper;
+}
+
+// TODO: use graasp-ui when window conflict is resolved
+// meanwhile we copy the definitions here
+export function usePlatformNavigation(
+  hostsMapper: HostsMapper,
+  itemId?: string,
+) {
+  return (platform: Platform) => {
+    const url = hostsMapper[platform]?.(itemId);
+    const href = url ?? '#';
+    return {
+      onClick: () => redirect(href),
+      onMouseDown: (event: React.MouseEvent) => {
+        if (event.button !== 1 || url === undefined) {
+          return;
+        }
+        window.open(href, '_blank');
+      },
+    };
+  };
+}
+
+export const APP_NAVIGATION_PLATFORM_SWITCH_ID = 'appNavigationPlatformSwitch';
+export const APP_NAVIGATION_PLATFORM_SWITCH_BUTTON_IDS = {
+  [Platform.Builder]: 'appNavigationPlatformSwitchButtonBuilder',
+  [Platform.Player]: 'appNavigationPlatformSwitchButtonPlayer',
+  [Platform.Library]: 'appNavigationPlatformSwitchButtonLibrary',
+  [Platform.Analytics]: 'appNavigationPlatformSwitchButtonAnalytics',
+};
+
+// small converter for HOST_MAP into a usePlatformNavigation mapper
+export const platformsHostsMap = defaultHostsMapper({
+  [Platform.Builder]: HOST_MAP.builder,
+  [Platform.Player]: HOST_MAP.player,
+});
+
+const linkStyles = {
+  textDecoration: 'none',
+  color: 'inherit',
+  display: 'flex',
+  alignItems: 'center',
+};
+
+const StyledLink = styled(Link)(() => linkStyles);
+
+interface HeaderNavigationProps {
+  rootId?: string;
+  topItemName?: string;
```

Comment:
What do you want to use this prop for ? It does not seem to get a value where this component is called.

Reply:
Good catch! Oops it was a copy paste from the component in Builder, but the prop is not used in LIbrary  👀